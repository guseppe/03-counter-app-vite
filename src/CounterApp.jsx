import {useState} from 'react'
import PropTypes from 'prop-types'


// const handleAdd = (event)=>{   
//     console.log(event);
// }

export const CounterApp = ({value}) => {
    console.log('render');
    const [counter, setCounter] = useState(value);
    const handleAdd = (event)=>{
       setCounter(counter + 1);
       //setCounter((c) => c + 1);
    }
    const handleSubstract = (event) => setCounter(counter - 1);
    const handleReset = (event) => setCounter(value);
    // const handleAdd = (event, newValue)=>{   //onClick={(event) => handleAdd(event, 'Hola')}
    //     console.log(newValue);
    // }
    return (
        <>
            <h1>Counter App</h1>
            <h2> {counter} </h2>
            <button onClick={ handleAdd }>
                +1
            </button>
            <button onClick={ handleSubstract }>-1</button>
            <button onClick={ handleReset }>reset</button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}