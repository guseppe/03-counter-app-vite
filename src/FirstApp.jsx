import PropTypes from 'prop-types'
//const newMessage = 'Guseppe';
//const newMessage = [1,2,3,4];
const newMessage = {  //no se puede renderizar el objecto completo, puedo hacer con los propiedades o usar JSON.stringify()
  message: 'Guseppe',
  title: 'Rodriguez',
};
const getMessage = ()=>{
  return 'Hola Mundo';
}
// export const FirstApp = ({title = 'default value', subtitle}) => {
  export const FirstApp = ({title, subtitle}) => {
  //const newMessage = 'Guseppe'; //se puede colocar fuera (no es un elemnto global, pertenece a un mismo modulo) si la expresion no necesita ser recalculada al renderizar nuevamente
  // console.log(title); //es mas habitual desestructurar las props
  return (
    <> {/*Uso de un Fragment implicito*/}
      <h1>{title}</h1> 
      {/* <h1>{props.title}</h1> */}
      {/* <h1>{newMessage.message}</h1> */}
      {/* <h1>{getMessage()}</h1> */}
      {/* <code>{JSON.stringify(newMessage)}</code> */}
      <p>{subtitle}</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.number
}
FirstApp.defaultProps = {
  title: 'No hay titulo',
  subtitle: 'No hay subtitulo'
}