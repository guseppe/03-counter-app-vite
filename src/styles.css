html, body{
    background-color: #21232A;
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem; /*rem para que sea uniforme en diferentes pantallasZ*/
    padding: 70px;
}
button{
    padding: 15px;
    font-size: 1.3rem;
    margin-right: 5px;
}