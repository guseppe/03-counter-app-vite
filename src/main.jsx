import React from 'react';
import ReactDOM from 'react-dom/client';

import {HelloWorldApp} from './HelloWorldApp'
import {FirstApp} from './FirstApp'
import './styles.css'
import { CounterApp } from './CounterApp';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {/* <FirstApp title='Guseppe' subtitle={123}/> */}
        {/* <FirstApp subtitle={123}/>  */}
        {/* <FirstApp/>  */}
        <>
            {/* <CounterApp value={12}/> */}
            <FirstApp/>
        </>
    </React.StrictMode>
);