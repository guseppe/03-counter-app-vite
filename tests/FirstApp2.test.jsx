import { getByText, render, screen } from "@testing-library/react";
import {FirstApp} from '../src/FirstApp'

describe('Pruebas en <FirstApp/>', () => { 
    const title = 'Hola soy goku';
    const subtitle = 'este es el subtitulo';
    test('debe hacer match con el snapshot', () => { 
        const {container} = render(<FirstApp title={title}/>);
        expect(container).toMatchSnapshot();
    })
    test('debe mostrar titulo en <h1>', () => { 
        render(<FirstApp title={title} />);
        //screen.debug();
        //expect(screen.getByText(title)).toBeTruthy(); 
        //expect(screen.getByText(title)).not.toBeTruthy(); que no exista
        const h1 = screen.getByRole('heading',{level:1});
        expect(h1.innerHTML).toContain(title);
        // const div = screen.getByTestId('titleId');
        // expect(div.innerHTML).toBe(title);

    })
    test('debe contener un subtitulo enviado', () => { 
        render(<FirstApp title={title} subtitle={subtitle} />);
        expect(screen.getAllByText(subtitle).length).toBe(2);
        expect(screen.getAllByText(subtitle)[0].innerHTML).toBe(subtitle);
        
    })
})
