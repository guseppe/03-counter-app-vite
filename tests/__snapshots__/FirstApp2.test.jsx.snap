// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> debe hacer match con el snapshot 1`] = `
<div>
   
  <h1
    data-testid="titleId"
  >
    Hola soy goku
  </h1>
  <p>
    no hay subtitulo
  </p>
  <p>
    no hay subtitulo
  </p>
</div>
`;
